docs/DeveloperGuide.md

```diff
@@ -1,18 +1,616 @@
+
+The program then checks if there are any entries upcoming within 5 days from the current date, and prints the entries out
+as reminders.
+
+1. Main code calls `MenuPrinter#printReminders()`, which in turn calls 
+`ReminderListGenerator#generateListOfRemindersAsStrings()`. 
+1. `ReminderListGenerator` checks the current date, and calculates the day of month which is 5 days from current date.
+This is stored in `dayToRemindUntil`.
+1. `ReminderListGenerator` then checks if `dayToRemindUntil` is after the last day of the current month. If it is,
+then the reminder timeframe will overflow to the next month. 
+    
+    For example:
+    * Current date is 29th October. There are 31 days in October. 5 days after today is 34th, 
+    which is beyond last day of October.
+    * Reminder timeframe will overflow to next month, until 3rd of November
+
+1. If it has overflown, set `isOverflowToNextMonth` to true. Subtract the last day of month from `dayToRemindUntil`.
+The new value of `dayToRemindUntil` is the day of next month that the reminder timeframe extends to.
+
+    For example:
+    * Continuing from example earlier, `dayToRemindUntil = 34`.
+    * `dayToRemindUntil -= NUM_DAYS_IN_OCT`, i.e. 34 - 31
+    * `dayToRemindUntil = 3`, representing that the reminder timeframe extends to 3rd of November
+1. `ReminderListGenerator` then grabs the entries within the reminder timeframe from the list of all recurring entries.
+    * If `isOverflowToNextMonth == true`, it will grab all entries from `currentDay` to `lastDayOfMonth` 
+    and all entries from `1` (1st day of next month) to `dayToRemindUntil`
+    * Else, it will simply grab all entries from `currentDay` to `dayToRemindUntil`
+
+1. Lastly, the list of entries will be converted to a formatted String to be displayed as reminders, and passed back
+to `MenuPrinter`, who will pass it to `UiManager` to print.
+
+The sequence diagram below shows how it works:
+
+![](uml_images/recurringtracker/images/reminderSeqDiagram.png)
+
+
+
+#### Feature 2: FinanceTools
+FinanceTools consists of the following features
+1. Simple Interest Calculator
+2. Yearly/Monthly Compound Interest Calculator
+3. Cashback Calculator
+4. Miles Credit Calculator
+6. Account Storage
+7. Command and Calculation History
+
+##### Simple Interest Calculator
+Simple Interest Calculator is facilitated by ```SimpleIntest``` class. It allows user to calculate interest earned.
+When user inputs ```simple``` as a command, ```handleSimpleInterest``` from ```Handler``` class will handle user
+inputted parameters. The calculation is done by ```SimpleInterest``` class. The result is outputted in
+```FinanceTools.main()```.
+<br />
+
+__Parameters__
+* ```/a``` - Amount (Mandatory)
+* ```/r``` - Interest Rate (Mandatory)
+
+The following class diagram shows how the Simple Interest Calculator feature works:
+<br />
+![ClassDiagram](uml_images/financetools/SimpleInterest/SimpleInterestClassDiagram.png)
```

Comment:
Perhaps you would like to take a look at this issue in the class forum about icons in PUML and consider the changes for the diagrams drawn in the following few sections:
https://github.com/nus-cs2113-AY2021S1/forum/issues/112

Reply:
Thanks a lot for this! Missed it inititally